### PR TITLE
Remove rid spec desktop builds

### LINKED ISSUE #19949: Avoid having runtime specific assets when they aren't necessary.
Due to an issue in nuget which will hopefully be address with https://github.com/NuGet/Home/issues/5192 some of our packages end up with the wrong asset being selected when we have both runtime and lib assets in the package. For some of our packages that isn't necessary.

Example: System.Security.Permissions has a runtime\net461 asset and a lib\netstandard2.0 asset and when a netcoreapp2.0 project has a fallback of net461 we will prefer the net461 asset over the netstandard2.0 asset which will not work on netcoreapp2.0. See https://github.com/dotnet/corefx/issues/19929. 

We should do an audit of other packages in this situation to avoid it.

### PR DESCRIPTION
**Pull request by ericstj**
Remove RID from desktop projects where it is not needed.

Fixes #19949

### CODE REVIEW

**Comment by ericstj** [comment]
I wrote some hacky code in build tools to test for this, which is how I found it.

What the test does is examine all supported TFM/RID combinations and evaluate the applicable assets.  Then it removes every asset that was applicable and determines applicability.  If it finds another dll that came from the same source location then it flags the path as redundant.

I'll see about polishing that test and adding it to build tools in the future.

File: src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj [hunk 1]
```diff
@@ -14,6 +13,9 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Unix-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Windows_NT-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Release|AnyCPU'" />
+  <!-- Default configurations to help VS understand the options -->
```
Comment (weshaggard) [inline src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj]: rogue comment we can probable delete.

File: src/System.Reflection.TypeExtensions/src/System.Reflection.TypeExtensions.csproj [hunk 2]
```diff
@@ -28,6 +27,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uapaot-Windows_NT-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uapaot-Windows_NT-Release|AnyCPU'" />
+  <!-- Default configurations to help VS understand the configurations -->
```
Comment (weshaggard) [inline src/System.Reflection.TypeExtensions/src/System.Reflection.TypeExtensions.csproj]: rogue comment

File: src/System.Security.Permissions/src/System.Security.Permissions.csproj [hunk 3]
```diff
@@ -8,11 +8,11 @@
...
     <PackageAsRefAndLib Condition="'$(TargetGroup)' == 'netfx'">true</PackageAsRefAndLib>
   </PropertyGroup>
-  <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Release|AnyCPU'" />
+  <!-- Default configurations to help VS understand the configurations -->
```
Comment (weshaggard) [inline src/System.Security.Permissions/src/System.Security.Permissions.csproj]: rogue comment

**Comment by ericstj** [comment]
I cleaned up all the rogue comments.

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Release x64 Build and Test

**Pull request merged**